File: Algorithms/Implementation/encryption.js
// Complete the encryption function below.
function encryption(s) {
    let result = ""

    let rows = Math.floor(Math.sqrt(s.length))
    let cols;
    if(Math.ceil(Math.sqrt(s.length)) == parseInt(rows)){
        cols = parseInt(rows)
    } else {
        cols = parseInt(rows) + 1
    }

    for ( let i = 0; i < cols; i++) {
        let p = i
        while(p < s.length){
            result += s[p]
            p += cols
        }
        result += ' '
    }

    return result;

}

File: Algorithms/Implementation/encryption.py
import math
# Complete the encryption function below.
def encryption(s):
    s = s.replace(" ", "")
    sqr = len(s) ** 0.5  # which is square root
    col = int(sqr) if math.ceil(sqr) == int(sqr) else int(sqr)+1

    word = ''
    for i in range(col):
        p = i
        while p < len(s):
            word = word + s[p]
            p += col
        word = word + ' '

    return word